haystack/knowledge_graph/base.py

```diff
@@ -0,0 +1,15 @@
+from typing import Optional
+
+from haystack import BaseComponent
+
+
+class BaseKnowledgeGraph(BaseComponent):
+    outgoing_edges = 1
+
+    def run(self, sparql_query: str, index: Optional[str] = None, **kwargs): # type: ignore
+        result = self.query(query=sparql_query, index=index)
+        output = {"sparql_result": result, **kwargs}
+        return output, "output_1"
+
+    def query(self, query: str, index: Optional[str] = None):
```

Comment:
Is the `query` here supposed to be a SPARQL query? If that's the case, we could consider replacinging the parameter name to `sparql_query`.

Reply:
Agreed, we now have "query" as in the other components (instead of question) and "sparql_query" for queries that can be directly executed on the knowledge graph without any translation.